src/trace.rs

```diff
@@ -39,3 +53,137 @@ pub fn init(color: bool, json: bool, levels: &str) {
 pub fn current_span() -> Span {
     Span::current()
 }
+
+pub fn subscribe() -> Option<Receiver<Event>> {
+    SENDER.get().map(|sender| sender.subscribe())
+}
+
+struct BroadcastSubscriber<F> {
+    sender: Sender<Event>,
+    formatter: F,
+}
+
+impl<F: Subscriber + 'static> Subscriber for BroadcastSubscriber<F> {
+    #[inline]
+    fn enabled(&self, metadata: &Metadata<'_>) -> bool {
+        self.formatter.enabled(metadata)
+    }
+
+    #[inline]
+    fn new_span(&self, span: &tracing::span::Attributes<'_>) -> Id {
+        self.formatter.new_span(span)
+    }
+
+    #[inline]
+    fn record(&self, span: &Id, record: &tracing::span::Record<'_>) {
+        self.formatter.record(span, record)
+    }
+
+    #[inline]
+    fn record_follows_from(&self, span: &Id, follows: &Id) {
+        self.formatter.record_follows_from(span, follows)
+    }
+
+    #[inline]
+    fn event(&self, event: &tracing::Event<'_>) {
+        if self.sender.receiver_count() > 0 {
+            let _ = self.sender.send(event.into()); // Ignore errors
+        }
+        self.formatter.event(event)
+    }
+
+    #[inline]
+    fn enter(&self, span: &Id) {
+        self.formatter.enter(span)
+    }
+
+    #[inline]
+    fn exit(&self, span: &Id) {
+        self.formatter.exit(span)
+    }
+
+    #[inline]
+    fn current_span(&self) -> span::Current {
+        self.formatter.current_span()
+    }
+
+    #[inline]
+    fn clone_span(&self, id: &Id) -> Id {
+        self.formatter.clone_span(id)
+    }
+
+    #[inline]
+    fn try_close(&self, id: Id) -> bool {
+        self.formatter.try_close(id)
+    }
+
+    #[inline]
+    fn register_callsite(&self, meta: &'static Metadata<'static>) -> Interest {
+        self.formatter.register_callsite(meta)
+    }
+}
+
+impl From<&tracing::Event<'_>> for Event {
+    fn from(event: &tracing::Event<'_>) -> Self {
+        let now = chrono::Utc::now();
+        let mut maker = MakeLogEvent::default();
+        event.record(&mut maker);
+
+        let meta = event.metadata();
+        let mut data = BTreeMap::<String, Value>::new();
+        data.insert("name".into(), meta.target().to_string().into());
+        data.insert("target".into(), meta.target().to_string().into());
+        data.insert("level".into(), meta.level().to_string().into());
+        data.insert(
+            "module_path".into(),
+            meta.module_path().map(|mp| mp.to_string()).into(),
+        );
+        data.insert(
+            "kind".into(),
+            if meta.is_event() {
+                "event".to_string().into()
+            } else if meta.is_span() {
+                "span".to_string().into()
+            } else {
+                Value::Null
+            },
+        );
+
+        maker.0.insert("metadata".into(), data.into());
+        maker.0.insert("timestamp".into(), now.into());
+
+        let log: LogEvent = maker.0.into_iter().collect();
```

Comment:
It might be more efficient to build `LogEvent`s here directly.

Reply:
There's probably a better way. I'll investigate.